packages/demos/src/start-server.ts

```diff
@@ -70,14 +59,22 @@ function createDocumentHtml(
 
 export async function startServer(
   webpackConfigs: webpack.Configuration[],
-  renderApp: AppRenderer | undefined,
   demoName?: string
 ): Promise<Server> {
   const port = await getPort(demoName ? {port: 3000} : undefined);
   const app = express();
 
+  for (const compiler of webpack(webpackConfigs).compilers) {
+    const serverSideRender =
+      compiler.options.output?.filename === nodeIntegratorFilename;
```

Comment:
from my point of view multiple dedicated webpack configs are better. Either two arrays for client and server or an extra wrapper with boolean property. This checking of the fileName is an implicit dependency, that is brittle. If more than two webpacks are needed

Reply:
I'm not sure I completely understand what you are suggesting here, since there can only be one node integrator per demo. But you do have a point that hard coding the node integrator file name might not be sich a good idea. I tried to improve on this in #639. Please have a look.